site/source/docs/api_reference/vr.h.rst

```diff
@@ -0,0 +1,378 @@
+	:returns: handle for a VR display.
+	:rtype: VRDisplayHandle
+
+.. c:function:: char* emscripten_vr_get_display_name(VRDisplayHandle handle)
+
+	Get a user-readable name which identifies the VR display.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: name of the VR display or `0 (NULL)` if the handle is invalid.
+	:rtype: char*
+
+.. c:function:: bool emscripten_vr_get_display_connected(VRDisplayHandle handle)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: `true` if the display is connected, `false` otherwise or when
+		the handle is invalid.
+	:rtype: bool
+
+.. c:function:: bool emscripten_vr_get_display_presenting(VRDisplayHandle handle)
+
+	See also :c:func:`emscripten_vr_request_present`.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:returns: `true` if the display is currently presenting, `false` otherwise
+		or when the handle is invalid.
+	:rtype: bool
+
+.. c:function:: int emscripten_vr_get_display_capabilities(VRDisplayHandle handle, VRDisplayCapabilities* displayCaps)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VRDisplayCapabilities displayCaps: receives capabilities of the VR display.
+	:returns: |display-function-return-doc|
+	:rtype: bool
+
+.. c:function:: int emscripten_vr_get_eye_parameters(VRDisplayHandle handle, VREye whichEye, VREyeParameters* eyeParams)
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|
+	:param VREye whichEye: which eye to query parameters for.
+	:param VREyeParameters eyeParam: receives the parameters for requested eye.
+	:returns: |display-function-return-doc|
+	:rtype: bool
+
+Render Loop
+===========
+
+In contrast to the usual emscripten main loop (see :ref:`emscripten-h-browser-execution-environment`),
+VR displays require their own rendering loop which is independent from the main loop. The rendering
+loop can be set per display and will act like a main loop with timing mode ``EM_TIMING_RAF`` until the
+display is requested to present, as of which it will run at the VR display's refresh rate.
+
+.. c:function:: void emscripten_vr_set_display_render_loop(VRDisplayHandle handle, em_vr_callback_func callback)
+
+	Set a C function as the per frame rendering callback of a VR display.
+
+	:param VRDisplayHandle handle: |display-handle-parameter-doc|: id of the display to set the render loop for.
+	:param em_vr_callback_func callback: C function to set as per frame rendering callback.
+	:rtype: |display-function-return-doc|
+
+.. tip:: There can be only *one* render loop function per VR display. To change the render loop function, first :c:func:`cancel <emscripten_vr_cancel_display_render_loop>` the current loop, and then call this function to set another.
+
+.. c:function:: void emscripten_vr_set_display_render_loop_arg(VRDisplayHandle handle, em_vr_callback_func callback, void* arg)
```

Comment:
Owch, here we followed the same paradigm as the original set main loop. That is great, though the multiple overloads there have been kind of historical, so we want to redesign this at some point.

Reply:
Is there already a new design? I'd be happy to simply remove the `emscripten_set_display_render_loop` function (without the userData).